src/DssSpell.sol

```diff
@@ -45,51 +49,100 @@ contract DssSpellAction is DssAction, DssSpellCollateralOnboardingAction {
-
-        DssExecLib.sendPaymentFromSurplusBuffer(FLIP_FLOP_FLAP_WALLET,  12_000);
-        DssExecLib.sendPaymentFromSurplusBuffer(ULTRASCHUPPI_WALLET,    12_000);
-        DssExecLib.sendPaymentFromSurplusBuffer(FEEDBLACK_LOOPS_WALLET, 12_000);
-        DssExecLib.sendPaymentFromSurplusBuffer(MAKERMAN_WALLET,        10_929);
-        DssExecLib.sendPaymentFromSurplusBuffer(ACRE_INVEST_WALLET,      9_347);
-        DssExecLib.sendPaymentFromSurplusBuffer(MONETSUPPLY_WALLET,      8_626);
-        DssExecLib.sendPaymentFromSurplusBuffer(JUSTIN_CASE_WALLET,      7_522);
-        DssExecLib.sendPaymentFromSurplusBuffer(GFX_LABS_WALLET,         6_607);
-        DssExecLib.sendPaymentFromSurplusBuffer(DOO_WALLET,                351);
-        DssExecLib.sendPaymentFromSurplusBuffer(FLIPSIDE_CRYPTO_WALLET,    265);
-
-        DssExecLib.setChangelogVersion("1.12.1");
+        // onboardNewCollaterals();
+
+        // -------------------------- Cure + New End ---------------------------
+        require(EndAbstract(MCD_END_OLD).live() == 1, "system-is-caged");
+        DssExecLib.setValue(MCD_END, "wait", EndAbstract(MCD_END_OLD).wait());
+        DssExecLib.setContract(MCD_END, "cure", MCD_CURE);
+        DssExecLib.authorize(MCD_CURE, MCD_END);
+
+        require(MCD_VAT == EndAbstract(MCD_END_OLD).vat(), "sanity-error-vat");
+        DssExecLib.setContract(MCD_END, "vat", MCD_VAT);
+        DssExecLib.authorize(MCD_VAT, MCD_END);
+        DssExecLib.deauthorize(MCD_VAT, MCD_END_OLD);
+
+        require(MCD_VOW == EndAbstract(MCD_END_OLD).vow(), "sanity-error-vow");
+        DssExecLib.setContract(MCD_END, "vow", MCD_VOW);
+        DssExecLib.authorize(MCD_VOW, MCD_END);
+        DssExecLib.deauthorize(MCD_VOW, MCD_END_OLD);
+
+        require(MCD_CAT == EndAbstract(MCD_END_OLD).cat(), "sanity-error-cat");
+        DssExecLib.setContract(MCD_END, "cat", MCD_CAT);
+        DssExecLib.authorize(MCD_CAT, MCD_END);
+        DssExecLib.deauthorize(MCD_CAT, MCD_END_OLD);
+
+        require(MCD_DOG == EndAbstract(MCD_END_OLD).dog(), "sanity-error-dog");
+        DssExecLib.setContract(MCD_END, "dog", MCD_DOG);
+        DssExecLib.authorize(MCD_DOG, MCD_END);
+        DssExecLib.deauthorize(MCD_DOG, MCD_END_OLD);
+
+        require(MCD_SPOT == EndAbstract(MCD_END_OLD).spot(), "sanity-error-spot");
+        DssExecLib.setContract(MCD_END, "spot", MCD_SPOT);
+        DssExecLib.authorize(MCD_SPOT, MCD_END);
+        DssExecLib.deauthorize(MCD_SPOT, MCD_END_OLD);
+
+        require(MCD_POT == EndAbstract(MCD_END_OLD).pot(), "sanity-error-pot");
+        DssExecLib.setContract(MCD_END, "pot", MCD_POT);
+        DssExecLib.authorize(MCD_POT, MCD_END);
+        DssExecLib.deauthorize(MCD_POT, MCD_END_OLD);
+
+        DssExecLib.setContract(MCD_ESM, "end", MCD_END);
+        DssExecLib.authorize(MCD_END, MCD_ESM);
+        DssExecLib.deauthorize(MCD_END_OLD, MCD_ESM);
```

Comment:
@gbalabasquer what are your thoughts on Deauthorizing the Pause Proxy from the OLD End? I think it could help us in the future since if we try to update the old one it would revert, but obviously we can only do that if we are fully done with it

Reply:
We haven't done with any contract that we have offboarded in the past. So I'd say we keep the permission for now until more thinking can be done. Honestly it would be very hard doing things in an old contract as we always interact with the chainlog. That's why it is important just to hardcode address in the spell from new contracts but not existing ones.